data/mods/ssb/moves.js

```diff
@@ -3285,42 +3285,56 @@ let BattleMovedex = {
 		type: "Electric",
 	},
 	// Saburo
-	soulbend: {
+	magmaore: {
 		accuracy: true,
 		basePower: 0,
 		category: "Status",
-		desc: "Raises the user's Attack and Speed by 1 stage. Has a 10% chance to summon either Reflect or Light Screen.",
-		shortDesc: "Atk, Spe +1; 10% chance to set one screen.",
-		id: "soulbend",
-		name: "Soulbend",
+		desc: "Sets up a hazard on the opposing side of the field, damaging each opposing Pokemon that switches in. Fails if the effect is already active on the opposing side. Foes lose 1/16 of their maximum HP, rounded down. Has a 10% chance to freeze the foe. Causes hail on the field Can be removed from the opposing side if any opposing Pokemon uses Rapid Spin or Defog successfully, or is hit by Defog.",
+		shortDesc: "Hurts foes on switch-in; 10% freeze; sets hail.",
+		id: "magmaore",
+		name: "Magma Ore",
 		isNonstandard: "Custom",
-		pp: 10,
+		pp: 20,
 		priority: 0,
-		flags: {snatch: 1, mirror: 1},
+		flags: {reflectable: 1},
 		onTryMove() {
 			this.attrLastMove('[still]');
 		},
 		onPrepareHit(target, source) {
-			this.add('-anim', source, "Refresh", source);
-			this.add('-anim', source, "Geomancy", source);
+			this.add('-anim', source, 'Stealth Rock', target);
 		},
-		boosts: {
-			atk: 1,
-			spe: 1,
+		sideCondition: 'magmaore',
+		self: {
+			onHit(pokemon) {
+				pokemon.addVolatile('magmaore2');
+			},
 		},
-		secondary: {
-			chance: 10,
-			onHit(target, source) {
-				let result = this.random(2);
-				if (result === 0) {
-					source.side.addSideCondition('reflect', source);
-				} else {
-					source.side.addSideCondition('lightscreen', source);
+		effect: {
+			// this is a side condition
+			onStart(side) {
+				this.add('-message', 'Molten lava appears on the field!');
+			},
+			onSwitchIn(pokemon) {
+				let typeMod = this.clampIntRange(pokemon.runEffectiveness(this.getActiveMove('stealthrock')), -6, 6);
+				this.damage(pokemon.maxhp * Math.pow(2, typeMod) / 8);
+				if (!(pokemon.hp <= 0 || pokemon.fainted) && this.random(10) < 4) {
+					pokemon.trySetStatus('brn');
+					// @ts-ignore
+					pokemon.magmaOre = true;
```

Comment:
Isn't there a `pokemon.m` value we can use for flags like this?

Reply:
Do I just use that as pokemon.m.magmaOre?

Reply:
yes